src/utils/narrow.js

```diff
@@ -306,11 +306,16 @@ export const canSendToNarrow = (narrow: Narrow): boolean =>
 
 export const getNarrowFromMessage = (message: Message | Outbox, ownEmail: string) => {
   if (Array.isArray(message.display_recipient)) {
-    const recipient =
-      message.display_recipient.length > 1
-        ? message.display_recipient.filter(x => x.email !== ownEmail)
-        : message.display_recipient;
-    return groupNarrow(recipient.map(x => x.email));
+    if (message.display_recipient.length > 1) {
+      return groupNarrow(
+        message.display_recipient.filter(x => x.email !== ownEmail).map(x => x.email),
+      );
+    } else if (message.display_recipient.length === 1) {
+      return privateNarrow(message.display_recipient[0].email);
```

Comment:
> I have a change that unifies the two of them in one of my draft branches for refactoring `Narrow`, but it's atop a bunch of other bigger changes. I noticed this same confusion in this bit of code the other day when preparing #4329, and I think I've seen it somewhere else recently too. So it might be past time for me to try making just that unification on its own -- it should simplify any further Narrow-refactoring anyway.

OK, and sent this as #4330!

Turned out the `privateNarrow` / `groupNarrow` unification was actually at the very start of that branch. :slightly_smiling_face: But I went and included some other pieces as well.

Reply:
Great! I just reviewed and merged #4330; it LGTM.